MetaMask/metamask-extension
# fix: close network/accounts toasts when opening the Edit page

## **Description**



This PR fixes the issue where toasts persist after navigating between `Permissions view` > `Edit networks` or `Edit accounts`. The fix ensures the toast dismisses upon navigating to edit in the Permissions modal.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/29239?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/27808

## **Manual testing steps**

1. Connect a dapp and grant permissions.
2. Go to the Connection icon > Permissions view > Accounts > Edit.
3. Change the account selection.
4. Update and verify the toast confirms account permissions were updated.
5. Navigate to Networks > Edit.
6. Change the network selection.
7. Update and verify the toast confirms network permissions were updated.
8. Confirm no stale toast persists after each navigation.

## **Screenshots/Recordings**



### **Before**

![Brave Browser 2024-10-11 at 4 24 18 PM](https://github.com/user-attachments/assets/a011d199-e0cc-40a7-b059-16d808c18964)

### **After**

https://github.com/user-attachments/assets/6cdd3b9d-90ce-4435-aea8-cf990b8b985a


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g., pulled and built the branch, ran the app, tested the code being changed).
